src/System.Security.AccessControl/src/System/Security/AccessControl/Privilege.cs

```diff
@@ -471,28 +471,25 @@ private void ToggleState(bool enable)
                         this.tlsContents.IncrementReferenceCount();
                     }
 
-                    Interop.Advapi32.LUID_AND_ATTRIBUTES luidAndAttrs = new Interop.Advapi32.LUID_AND_ATTRIBUTES();
-                    luidAndAttrs.Luid = this.luid;
-                    luidAndAttrs.Attributes = enable ? Interop.Advapi32.SEPrivileges.SE_PRIVILEGE_ENABLED : Interop.Advapi32.SEPrivileges.SE_PRIVILEGE_DISABLED;
-
-                    Interop.Advapi32.TOKEN_PRIVILEGE newState = new Interop.Advapi32.TOKEN_PRIVILEGE();
+                    Interop.Advapi32.TOKEN_PRIVILEGE newState;
                     newState.PrivilegeCount = 1;
-                    newState.Privileges[0] = luidAndAttrs;
+                    newState.Privileges.Luid = this.luid;
+                    newState.Privileges.Attributes = enable ? Interop.Advapi32.SEPrivileges.SE_PRIVILEGE_ENABLED : Interop.Advapi32.SEPrivileges.SE_PRIVILEGE_DISABLED;
 
-                    Interop.Advapi32.TOKEN_PRIVILEGE previousState = new Interop.Advapi32.TOKEN_PRIVILEGE();
-                    uint previousSize = 0;
+                    Interop.Advapi32.TOKEN_PRIVILEGE previousState;
+                    uint previousSize;
 
                     //
                     // Place the new privilege on the thread token and remember the previous state.
                     //
 
-                    if (false == Interop.Advapi32.AdjustTokenPrivileges(
+                    if (!Interop.Advapi32.AdjustTokenPrivileges(
                                       this.tlsContents.ThreadHandle,
                                       false,
-                                      ref newState,
-                                      (uint)Marshal.SizeOf(previousState),
-                                      ref previousState,
-                                      ref previousSize))
+                                      &newState,
+                                      (uint)sizeof(Interop.Advapi32.TOKEN_PRIVILEGE),
+                                      &previousState,
+                                      &previousSize))
```

Comment:
Is previousSize used anywhere?  If not, could this just be null instead and remove the local?

Reply:
MSDN says that this argument can be null only if previousState is null as well. It is not the case here.